src/main/java/main/model/db/dao/project/UserDao.java

```diff
@@ -20,16 +25,39 @@ public UserDao() {
         remove = "{call REMOVE_USER(?)}";
     }
 
-    public UserDto IsAuthorized(String sessionId) throws AqualityException {
+    /**
+     * Check id session or token is valid
+     * @param sessionHash Base64 String with Session ID = {username}:{uuid}:{creationDate} or project:{projectId}:{token}
+     * @return User for session
+     * @throws AqualityException error about authorization status
+     */
+    public UserDto GetAuthorizedUser(String sessionHash) throws AqualityException {
         Base64 base64 = new Base64();
+        String sessionId = StringUtils.newStringUtf8(base64.decode(sessionHash));
+        boolean isApiToken = sessionId.startsWith("project");
+        if(isApiToken){
+            return IsAuthorizedToken(sessionId);
+        }
+
+        return IsAuthorizedUser(sessionId, sessionHash);
+    }
+
+    /**
+     * Check If User Authorized
+     * @param sessionId Session ID = {username}:{uuid}:{creationDate}
+     * @param sessionHash Hash of Session ID = {username}:{uuid}:{creationDate}
+     * @return Authorized User
+     * @throws AqualityException error about authorization status
+     */
+    private UserDto IsAuthorizedUser(String sessionId, String sessionHash) throws AqualityException {
+        String[] strings = sessionId.split(":");
         DateUtils dates = new DateUtils();
-        String[] strings = StringUtils.newStringUtf8(base64.decode(sessionId)).split(":");
         UserDto user = new UserDto();
         user.setUser_name(strings[0]);
         List<UserDto> users = searchAll(user);
         if(users.size() > 0){
             user = users.get(0);
-            if(!user.getSession_code().equals(sessionId)){
+            if(!user.getSession_code().equals(sessionHash)){
```

Comment:
can we invert logic: at first PASS way (all is all right) and then error processing?

Reply:
fixed